sensor/common/registry/registry_store_test.go

```diff
@@ -43,186 +50,264 @@ func alwaysFailCheckTLS(_ context.Context, _ string) (bool, error) {
-	require.NoError(t, regStore.UpsertRegistry(ctx, "qa", "172.99.12.11:5000", dce))
+
+	regStore.UpsertSecret("qa", fakeSecretName, dc, noServiceAcctName)
 
 	img := &storage.ImageName{
 		Registry: "image-registry.openshift-image-registry.svc:5000",
 		Remote:   "qa/nginx",
 		Tag:      "nginx:1.18.0",
 		FullName: "image-registry.openshift-image-registry.svc:5000/qa/nginx:1.18.0",
 	}
-	reg, err := regStore.GetRegistryForImageInNamespace(img, "qa")
+	regs, err := regStore.GetPullSecretRegistries(img, "qa", nil)
 	require.NoError(t, err)
-	assert.Equal(t, "image-registry.openshift-image-registry.svc:5000", reg.Config(ctx).RegistryHostname)
+	require.Len(t, regs, 1)
+	assert.Equal(t, img.GetRegistry(), regs[0].Config(bgCtx).RegistryHostname)
 
 	img = &storage.ImageName{
 		Registry: "image-registry.openshift-image-registry.svc.local:5000",
 		Remote:   "qa/nginx",
 		Tag:      "nginx:1.18.0",
 		FullName: "image-registry.openshift-image-registry.svc.local:5000/qa/nginx:1.18.0",
 	}
-	reg, err = regStore.GetRegistryForImageInNamespace(img, "qa")
+
+	regs, err = regStore.GetPullSecretRegistries(img, "qa", nil)
 	require.NoError(t, err)
-	assert.Equal(t, "image-registry.openshift-image-registry.svc.local:5000", reg.Config(ctx).RegistryHostname)
+	require.Len(t, regs, 1)
+	assert.Equal(t, img.GetRegistry(), regs[0].Config(bgCtx).RegistryHostname)
 
 	img = &storage.ImageName{
 		Registry: "172.99.12.11:5000",
 		Remote:   "qa/nginx",
 		Tag:      "nginx:1.18.0",
 		FullName: "172.99.12.11:5000/qa/nginx:1.18.0",
 	}
-	reg, err = regStore.GetRegistryForImageInNamespace(img, "qa")
+	regs, err = regStore.GetPullSecretRegistries(img, "qa", nil)
 	require.NoError(t, err)
-	assert.Equal(t, "172.99.12.11:5000", reg.Config(ctx).RegistryHostname)
+	require.Len(t, regs, 1)
+	assert.Equal(t, img.GetRegistry(), regs[0].Config(bgCtx).RegistryHostname)
 }
 
 // TestRegistryStore_SpecificNamespace tests interactions with the registry store
 // using an explicitly provided namespace (vs. inferred)
 func TestRegistryStore_SpecificNamespace(t *testing.T) {
-	ctx := context.Background()
-	regStore := NewRegistryStore(alwaysInsecureCheckTLS)
+	t.Run("SecretsByHost", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SensorPullSecretsByName, false)
+		regStore := specificNamespaceSubTest(t)
+
+		// no registry should exist based on img.Remote
+		regs, err := regStore.GetPullSecretRegistries(fakeImgName, "qa", nil)
+		assert.Error(t, err)
+		assert.Empty(t, regs)
+	})
+
+	t.Run("SecretsByName", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SensorPullSecretsByName, true)
+		regStore := specificNamespaceSubTest(t)
+
+		// no registry should exist based on img.Remote
+		regs, err := regStore.GetPullSecretRegistries(fakeImgName, "qa", nil)
+		assert.NoError(t, err)
+		assert.Empty(t, regs)
+	})
+}
+
+func specificNamespaceSubTest(t *testing.T) *Store {
```

Comment:
I think this name is throwing me off. At first I didn't see the difference between this and the previous one because the namespaces were the same, but I believe it's due to the fact the fake image does not have an associated namespace. Can we change the name of this to make that clearer?

Reply:
Looking at it a bit closer, these tests (the previous one and this one) do not differ anymore - the 'guess the namespace' function of the past is gone after we started passing the 'actual' namespace through the enricher. I did some cleanup on the tests, added comments, and consolidated a few of them.  I believe it's in a much better state now.